### PR TITLE
Expose supported extensions

### PR DESCRIPTION
**Pull request by vallsv**
A convenient way to expose supported format.

We also can talk about refactoring this thing using a descriptor class per codec. It can be very useful to provides more informations.

### CODE REVIEW

File: silx/io/fabioh5.py [hunk 1]
```diff
@@ -53,6 +53,36 @@
 _logger = logging.getLogger(__name__)
 
 
+_fabio_extensions = set([])
+
+
+def available_extensions():
```
Comment (t20100) [inline silx/io/fabioh5.py]: Why not calling it `supported_extensions` as the `silx.io` one?

File: silx/io/utils.py [hunk 2]
```diff
@@ -53,6 +53,38 @@
...
+        formats["NeXus files"] = set(["*.nx", "*.nxs", "*.h5", "*.hdf"])
+    formats["NeXus layout from spec files"] = set(["*.dat", "*.spec", "*.mca"])
+    if flat_formats:
+        try:
+            import fabioh5
+        except ImportError:
+            fabioh5 = None
+        if fabioh5 is not None:
+            formats["NeXus layout from fabio files"] = set(fabioh5.available_extensions())
+    if numpy is not None:
```
Comment (t20100) [inline silx/io/utils.py]: numpy should always be available

File: silx/io/__init__.py [hunk 3]
```diff
@@ -36,6 +36,10 @@
...
 from .utils import is_group
 from .utils import is_softlink
+from .utils import supported_extensions
 
 # avoid to import open with "import *"
-__all__ = ["save1D", "is_dataset", "is_file", "is_group", "is_softlink"]
+__all = locals().keys()
+__all = filter(lambda x: not x.startswith("_"), __all)
+__all = filter(lambda x: x != "open", __all)
+__all__ = list(__all)
```
Comment (t20100) [inline silx/io/__init__.py]: I personally preferred the previous `__all__` definition even if it needs to be maintained by hand.

Comment (vallsv) [reply silx/io/__init__.py]: This do what it is supposed to do. Filtering out open.
The previous way is not maintainable, and will grow for every.

File: silx/io/fabioh5.py (hunk 1 above)
Comment (vallsv) [reply silx/io/fabioh5.py]: It is not the same output, but yes, it can be called the same

**Comment by vallsv** [comment]
I updated the thing according to your comments. But it still use filters for the `__all__`.

Maybe it would be better to remove everything. `open` is not anymore a reserved word in Python3 if i am not wrong, and no body should use `import *`. Then this filters looks to be mainly useless and without use case.

**Pull request merged**